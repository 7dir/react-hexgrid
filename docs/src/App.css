@font-face {
    font-family: 'SauceCodePro Nerd Font';
    src: local("./SauceCodeProNerdFontMono-Black.ttf");
    format: ('truetype');
}

* {
    font-family: "SauceCodePro Nerd Font", serif;
}