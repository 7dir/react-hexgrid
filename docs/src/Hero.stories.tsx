import React from "react";
import type { Meta, StoryObj } from "@storybook/react";
import Hero from "./Hero"

const meta = {
    title: "Stories/Hero",
} satisfies Meta<typeof Hero>

export default meta;

type Story = StoryObj<typeof Hero>

export const Default: Story = {
    render: () => (
        <Hero />
    )
}

