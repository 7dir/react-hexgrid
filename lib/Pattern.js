import React from "react";
import Point from "./models/Point";
const defaultSize = new Point(10, 10);
/**
 * Defines a `<defs><pattern><image>` group (will not be rendered) in order to allow defining images.
 * The given id can be used on the `Hexagon` to render the image
 */
export function Pattern({ id, link, size = defaultSize }) {
    return (React.createElement("defs", null,
        React.createElement("pattern", { id: id, patternUnits: "objectBoundingBox", x: 0, y: 0, width: size.x, height: size.y },
            React.createElement("image", { xlinkHref: link, x: 0, y: 0, width: size.x * 2, height: size.y * 2 }))));
}
export default Pattern;
//# sourceMappingURL=Pattern.js.map