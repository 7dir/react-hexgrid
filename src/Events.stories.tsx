import React from "react"
import type { Meta, StoryFn } from "@storybook/react";
import { HexGrid, Layout, Hexagon, GridGenerator, Hex } from "./";

const meta = {
    title: "Stories/Events",
    component: Hexagon,
} satisfies Meta<typeof Hexagon>;

export default meta;

type Story = StoryFn<typeof Hexagon>;

