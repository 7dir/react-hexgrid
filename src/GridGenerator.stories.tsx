import React from "react";
import type { Meta, StoryObj } from "@storybook/react";

import Hexagon from "./Hexagon/Hexagon";
import { HexGrid, GridGenerator, Layout } from "./"

const meta = {
    title: "Stories/GridGenerator",
} satisfies Meta<typeof GridGenerator>

export default meta;

type Story = StoryObj<typeof GridGenerator>

export const Default: Story = {
    render: () => (
        <>
            <HexGrid width="100%" height="100%">
                <Layout>
                    {GridGenerator.hexagon(2).map((hex, i) => (
                        <Hexagon key={i} q={hex.q} r={hex.r} s={hex.s} cellStyle={{ fill: "red", stroke: "black", strokeWidth: "0.1" }} />
                    ))}
                </Layout>
            </HexGrid>
        </>
    )
}

