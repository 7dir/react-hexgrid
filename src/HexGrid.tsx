import React, { Component } from "react"
import PropTypes from "prop-types"

export type HexGridProps = {
  width?: string | number
  height?: string | number
  viewBox?: string
  children: React.ReactElement | React.ReactElement[]
}
export function HexGrid({
  width = 800,
  height = 600,
  viewBox = "-50 -50 100 100",
  children,
}: HexGridProps) {
  // static propTypes = {
  //   width: PropTypes.oneOfType([
  //     PropTypes.string.isRequired,
  //     PropTypes.number.isRequired,
  //   ]),
  //   height: PropTypes.oneOfType([
  //     PropTypes.string.isRequired,
  //     PropTypes.number.isRequired,
  //   ]),
  //   viewBox: PropTypes.string,
  //   children: PropTypes.node.isRequired
  // };

  // static defaultProps = {
  //   width: 800,
  //   height: 600,
  //   viewBox: "-50 -50 100 100"
  // }

  return (
    <svg
      className="grid"
      width={width}
      height={height}
      viewBox={viewBox}
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      {children}
    </svg>
  )
}
export default HexGrid
