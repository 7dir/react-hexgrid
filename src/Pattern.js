import React, { Component } from "react"
import PropTypes from "prop-types"
import HexUtils from "./HexUtils"
import Point from "./models/Point"
import { Size } from "./Layout"

export type PatternProps = {
  id: string
  link: string
  size?: Size
}
const defaultSize = new Point(10, 10)

export function Pattern({ id, link, size = defaultSize }: PatternProps) {
  // static propTypes = {
  //   id: PropTypes.string.isRequired,
  //   link: PropTypes.string.isRequired,
  //   size: PropTypes.object
  // };
  // static defaultProps = {
  //   size: new Point(10, 10)
  // };

  // const { id, link, size } = this.props;

  return (
    <defs>
      <pattern
        id={id}
        patternUnits="objectBoundingBox"
        x={0}
        y={0}
        width={size.x}
        height={size.y}
      >
        <image
          xlinkHref={link}
          x={0}
          y={0}
          width={size.x * 2}
          height={size.y * 2}
        />
      </pattern>
    </defs>
  )
}

export default Pattern
