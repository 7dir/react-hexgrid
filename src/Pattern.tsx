import React from "react"
import Point from "./models/Point"
import { Size } from "./Layout"

export type PatternProps = {
  id: string
  x?: number
  y?: number
  
  link?: string
  rc?: React.ReactElement
  // ReactElement<any, any>
  size?: Size
}
const defaultSize = new Point(10, 10)

/**
 * Defines a `<defs><pattern><image>` group (will not be rendered) in order to allow defining images.
 * The given id can be used on the `Hexagon` to render the image
 */
export function Pattern({ id, x = 0, y = 0, link, rc, size = defaultSize }: PatternProps) {
  return (
    <defs>
      <pattern
        id={id}
        patternUnits="objectBoundingBox"
        x={x}
        y={y}
        width={size.x}
        height={size.y}
      >
        {link && (<image
          xlinkHref={link}
          x={0}
          y={0}
          width={size.x * 2}
          height={size.y * 2}
        />)}
        {rc && (<>{rc}</>)}
      </pattern>
    </defs>
  )
}

export default Pattern
