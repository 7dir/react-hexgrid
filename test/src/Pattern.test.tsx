import React from "react"
import { render } from "@testing-library/react"

import { Pattern } from "../../src/Pattern"

test("Pattern should render correctly with default props", () => {
  const { container } = render(
    <svg>
      <Pattern
        id={"test1"}
        link={"http://example.com"}
        size={{ x: 100, y: 200 }}
      />
    </svg>,
  )
  expect(container.firstChild).toMatchSnapshot()
})

test("Pattern should render correctly with custom props", () => {
  const { container } = render(
    <svg>
      <Pattern
        id={"test2"}
        link={"http://example.com"}
        size={{ x: 200, y: 100 }}
      />
    </svg>,
  )
  expect(container.firstChild).toMatchSnapshot()
})
