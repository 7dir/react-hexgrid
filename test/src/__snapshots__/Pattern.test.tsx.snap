// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pattern should render correctly with custom props 1`] = `
<svg>
  <defs>
    <pattern
      height="100"
      id="test2"
      patternUnits="objectBoundingBox"
      width="200"
      x="0"
      y="0"
    >
      <image
        height="200"
        width="400"
        x="0"
        xlink:href="http://example.com"
        y="0"
      />
    </pattern>
  </defs>
</svg>
`;

exports[`Pattern should render correctly with default props 1`] = `
<svg>
  <defs>
    <pattern
      height="200"
      id="test1"
      patternUnits="objectBoundingBox"
      width="100"
      x="0"
      y="0"
    >
      <image
        height="400"
        width="200"
        x="0"
        xlink:href="http://example.com"
        y="0"
      />
    </pattern>
  </defs>
</svg>
`;
